dotnet/samples/GettingStartedWithAgents/Step6_DependencyInjection.cs

```diff
@@ -0,0 +1,127 @@
+                    Name = TutorName,
+                    Kernel = sp.GetRequiredService<Kernel>(),
+                });
+
+        // Create a service provider for resolving registered services
+        await using ServiceProvider serviceProvider = serviceContainer.BuildServiceProvider();
+
+        // If an application follows DI guidelines, the following line is unnecessary because DI will inject an instance of the KernelClient class to a class that references it.
+        // DI container guidelines - https://learn.microsoft.com/en-us/dotnet/core/extensions/dependency-injection-guidelines#recommendations
+        AgentClient agentClient = serviceProvider.GetRequiredService<AgentClient>();
+
+        // Execute the agent-client
+        await WriteAgentResponse("The sunset is very colorful.");
+        await WriteAgentResponse("The sunset is setting over the mountains.");
+        await WriteAgentResponse("The sunset is setting over the mountains and filled the sky with a deep red flame, setting the clouds ablaze.");
+
+        // Local function to invoke agent and display the conversation messages.
+        async Task WriteAgentResponse(string input)
+        {
+            this.WriteLine($"# {AuthorRole.User}: {input}");
+
+            await foreach (var content in agentClient.RunDemoAsync(input))
+            {
+                this.WriteLine($"# {content.Role} - {content.AuthorName ?? "*"}: '{content.Content}'");
+            }
+        }
+    }
+
+    private sealed class AgentClient([FromKeyedServices(TutorName)] ChatCompletionAgent agent)
+    {
+        private readonly AgentGroupChat _chat =
+            new()
+            {
+                ExecutionSettings =
+                    new()
+                    {
+                        // Here a TerminationStrategy subclass is used that will terminate when
+                        // the response includes a score that is greater than or equal to 70.
+                        TerminationStrategy = new ThresholdTerminationStrategy()
+                    }
+            };
+
+        public IAsyncEnumerable<ChatMessageContent> RunDemoAsync(string input)
+        {
+            // Create a chat for agent interaction.
+
+            this._chat.AddChatMessage(new ChatMessageContent(AuthorRole.User, input));
+
+            return this._chat.InvokeAsync(agent);
+        }
+    }
+
+    private record struct InputScore(int score, string notes);
+
+    private sealed class ThresholdTerminationStrategy : TerminationStrategy
+    {
+        protected override Task<bool> ShouldAgentTerminateAsync(Agent agent, IReadOnlyList<ChatMessageContent> history, CancellationToken cancellationToken)
+        {
+            string lastMessageContent = history[history.Count - 1].Content ?? string.Empty;
+
+            InputScore? result = JsonResultTranslator.Translate<InputScore>(lastMessageContent);
+
+            return Task.FromResult((result?.score ?? 0) >= 70);
```

Comment:
Wondering if making `70` value configurable will improve sample experience for users?

Reply:
Good point, at the least it will increase the discoverability of this logic / threshold.  Pushing a `const`

Reply:
Updated